source/MRMesh/MRFillContourByGraphCut.cpp

```diff
@@ -317,4 +332,13 @@ FaceBitSet fillContourLeftByGraphCut( const MeshTopology & topology, const std::
     return filler.fill();
 }
 
+FaceBitSet segmentByGraphCut( const MeshTopology& topology, const FaceBitSet& source, const FaceBitSet& sink, const EdgeMetric& metric )
+{
+    MR_TIMER
+    GraphCut filler( topology, metric );
+    filler.addFaces( source, true );
+    filler.addFaces( sink, false );
```

Comment:
better make a function `addFaces` that will take both source and sink argument. It will make it much simpler inside